File: client/src/app/core/entity/vaccineoffering.ts
import {VaccinationStage} from "./vaccinationstage";
import {Dose} from "./dose";

export class VaccineOffering{
  id:number
  dose:Dose
  vaccinationstage:VaccinationStage
  year:number
  month:number
  title?:string

  constructor(id:number,dose:Dose,vaccinationstage:VaccinationStage,year:number,month:number,title:string) {
    this.id = id;
    this.dose = dose;
    this.vaccinationstage = vaccinationstage;
    this.year = year;
    this.month = month;
    this.title = title;

  }
}

File: server/src/main/java/com/content/security/entity/Vaccineoffering.java
package com.content.security.entity;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
@Entity
@Table(name = "vaccineoffering")
public class Vaccineoffering {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", nullable = false)
    private Integer id;

    @NotNull
    @ManyToOne(fetch = FetchType.EAGER, optional = false)
    @JoinColumn(name = "dose_id", nullable = false)
    private Dose dose;

    @NotNull
    @ManyToOne(fetch = FetchType.EAGER, optional = false)
    @JoinColumn(name = "vaccinationstage_id", nullable = false)
    private Vaccinationstage vaccinationstage;

    @JsonIgnore
    @NotNull
    @ManyToOne(fetch = FetchType.EAGER, optional = false)
    @JoinColumn(name = "vaccine_id", nullable = false)
    private Vaccine vaccine;

    @Column(name = "year")
    private Integer year;

    @Column(name = "month")
    private Integer month;

    @Column(name = "title")
    private String title;

}